File: design-patterns/ts/state/order-status/order-check-state.ts
// order-check-state.ts

import OrderContext from "./order-context";
import OrderState from "./order-state";

class OrderCheckState extends OrderState {
    constructor(context: OrderContext) {
        super(context);
    }

    process(): void {
        // Write code to process the order
        console.log("Checking the order validity and other information");

        this.context.setState(this.context.getOrderInProgressState());
    }
}

export default OrderCheckState;

File: design-patterns/go/state/order_state/order_check_state.go
// order_check_state.go

package main

import "fmt"

type OrderCheckState struct {
	*OrderState
}

func NewOrderCheckState(context *OrderContext) (orderCheckState *OrderCheckState) {
	orderCheckState = &OrderCheckState{}
	orderCheckState.OrderState = NewOrderState(orderCheckState, context)
	return
}

func (orderCheckState *OrderCheckState) Process() {
	fmt.Println("Checking the order validity and other information")
	orderCheckState.context.SetState(orderCheckState.context.GetOrderInProgressState())
}
